backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -137,7 +138,25 @@ private void fetchCredentialsAndInitializeS3ClientWithTemporaryCredentials(Regio
     }
   }
 
-  private void initializeS3ClientWithTemporaryCredentials(Regions clientRegion) {
+  private synchronized void initializeS3ClientWithTemporaryCredentials(
```

Comment:
comment for conrado's review: synchronized to avoid multiple thread doing the operations in parallel